src/purifier/sanitation.js

```diff
@@ -264,6 +264,7 @@ const DENYLISTED_TAG_SPECIFIC_ATTRS = Object.freeze(
     'input': DENYLISTED_FIELDS_ATTR,
     'textarea': DENYLISTED_FIELDS_ATTR,
     'select': DENYLISTED_FIELDS_ATTR,
+    'template': ['shadowroot'],
```

Comment:
I think this covers templates rendered by templates, since the sanitizer runs on the contents of the template, not the template element itself. Is that what we need. I honestly don't fully understand the vector here.

Reply:
I think it's the same as this [thread](https://github.com/ampproject/amphtml/pull/30915#discussion_r514612315). Deduping into it.